pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -248,19 +249,27 @@ public boolean isPartialUpsertEnabled() {
     return _upsertMode == UpsertConfig.Mode.PARTIAL;
   }
 
+  /**
+   * Add or replace an immutable segment created by the REALTIME table.
+   * Skipping segment still in consuming state.
+   */
+  @Override
+  public void addOrReplaceSegment(String segmentName, IndexLoadingConfig indexLoadingConfig,
+      SegmentMetadata localMetadata, SegmentZKMetadata zkMetadata, boolean forceDownload)
+      throws Exception {
+    throw new UnsupportedOperationException();
```

Comment:
Why do we throw exception here?

Reply:
I was planning to add support for RT table in next PR, and make this one focus on refactoring existing logic. As reload...() APIs are to be extended with forceDownload flag, I'll double check if changes are still needed here.

Reply:
it looks like RealTime table doesn't support RefreshSegmentMsg [yet](https://github.com/apache/pinot/blob/master/pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java#L75) thus no chance to call addOrReplaceSegment() on RT table mgr.